### ISSUE TITLE
Log formatting improvement

### ISSUE THREAD

**Issue by MelReyCG** [GEOSX/GEOSX#2277]
**What is the requested feature?**
The log is hard to read by the users in our production environnement. So, in order to improve GeoSX usability, we need to improve the log formatting.

**Is your request related to a specific problem?**
- All messages aren't necessarily relevant and some of them are very repetitive,
- Numbers are sometimes too large (no thousand separators for large integers, too much precision for decimals),
- It's hard to retrieve warnings and errors among all the info messages,

**Describe the solution you'd like**
- [ ] Add a logLevel at the <Problem> level to control general log lines,
- [ ] Distribute current messages in log levels,
- [ ] Review the numbers formatting to improve readability (add thousand separators for large integers, review the number of significant digits),
- [ ] Add a "message type" header for each log lines (Warning, Error, Info...) with proper alignement,

**Describe alternatives you've considered**
Another solution would be to write another log for the users (and the current one would be intended for developpers).

**Comment by herve-gross** [comment]
@MelReyCG can you prepare a 15-min simple presentation before you code in a branch of what you think the log should look like? It will be easier for us to help you and give you advice.

**Comment by MelReyCG** [comment]
Sure, I will do this presentation when I will have a more precise definition of the log format desired by the users.

**Comment by TotoGaz** [comment]
Chat back with @jeannepellerin about this 😉

**Comment by TotoGaz** [comment]
You should also be aware of https://github.com/GEOSX/GEOSX/issues/1613 and https://github.com/GEOSX/GEOSX/issues/2298

**Comment by jeannepellerin** [comment]
@rrsettgast

**Comment by MelReyCG** [comment]
Any suggestion @paveltomin @CusiniM @castelletto1 @klevzoff ?

**Comment by paveltomin** [comment]
> Any suggestion @paveltomin @CusiniM @castelletto1 @klevzoff ?

Good initiative!
So many things, as already suggested a presentation/discussion session would be great.
General suggestion: keep it as simple as possible.

**Comment by herve-gross** [comment]
If I can chime in, it is important for logs to be human-readable, but we should also make machine-parsing (with Python, for instance) as straightforward as possible.

**Comment by TotoGaz** [comment]
> If I can chime in, it is important for logs to be human-readable, but we should also make machine-parsing (with Python, for instance) as straightforward as possible.

Yes, this is very true. We can do anything we want on the log as long as there's another source of information aside that is meant to be parsed.
Otherwise, tables like in https://github.com/GEOS-DEV/GEOS/pull/2984 will be a challenge. Maybe you think that we should use tools like https://docs.astropy.org/en/latest/io/ascii/read.html, but still that would commit us to using this python module.

**Comment by victorapm** [comment]
This is a nice initiative. If I may chime in as well, one option that makes parsing with Python easier and it is still human readable is Markdown, e.g.:
```
| Element no. | CoordX | CoordY | CoordZ             | Prev Element | Next Element |
|-------------|--------|--------|--------------------|--------------|--------------|
| 0           | 9.5    | 0.2    | 11.958333333333334 | 1            |              |
| 1           | 9.5    | 0.2    | 11.875             | 2            | 0            |
| 2           | 9.5    | 0.2    | 11.791666666666666 | 3            | 1            |
| 3           | 9.5    | 0.2    | 11.708333333333334 | 4            | 2            |
| 4           | 9.5    | 0.2    | 11.625             | 5            | 3            |
```

**Comment by MelReyCG** [comment]
I can see 3 strategies that could solve this need for machine & human readability :
1. As this work is already started by @paveltomin, we can dump all lines that is replaced by tables in a CSV file with a writeCSV option, waiting for hdf5 support later (new tables too).
2. We can keep the legacy logs just below the new human readable line, like I did with the timestep. In this case we will need to provide a table format that is easy to parse (like @victorapm suggested). We could add an option to show the _legacy logs_ (like `-l` / `--legacyLogs`).
3. We can provide a human readable log only with tables that are not too hard to parse (like the one @victorapm suggested), but I think that is the strategy that will need the more work to the log reader developers.

I will prepare a presentation with more insight on what I planned, so we will be able to talk about the strategy to implement.

@victorapm 
What do you think of this to support the title name, and to provide an easy to parse starting & ending ?

```
------------------------------------------------------------------------------------
|                             WellGeneration myWell                                |
|----------------------------------------------------------------------------------|
| Element no. | CoordX | CoordY | CoordZ             | Next Element | Prev Element |
|-------------|--------|--------|--------------------|--------------|--------------|
| 0           | 9.5    | 0.2    | 11.958333333333334 | 1            |              |
| 1           | 9.5    | 0.2    | 11.875             | 2            | 0            |
| 2           | 9.5    | 0.2    | 11.791666666666666 |              | 1            |
------------------------------------------------------------------------------------
```
